### PR TITLE
fixes #4863 - API V2 - add puppetclasses format style=list and module_name to base.json.rabl

### PR DESCRIPTION
**Pull request by isratrade**

### CODE REVIEW

**Comment by isratrade** [comment]
I need these in the API v2 for re-factoring the hostgroup form in angularjs

File: test/functional/api/v2/puppetclasses_controller_test.rb [hunk 1]
```diff
@@ -9,6 +9,16 @@ class Api::V2::PuppetclassesControllerTest < ActionController::TestCase
     assert_response :success
     puppetclasses = ActiveSupport::JSON.decode(@response.body)
     assert !puppetclasses.empty?
+    assert puppetclasses['results'].kind_of?(Hash)
+  end
+
+  # format=list
```
Comment (dLobatog) [inline test/functional/api/v2/puppetclasses_controller_test.rb]: Why this comment?

File: test/functional/api/v2/puppetclasses_controller_test.rb [hunk 2]
```diff
@@ -9,6 +9,16 @@ class Api::V2::PuppetclassesControllerTest < ActionController::TestCase
...
+  test "should get index with style=list" do
+    get :index, {:style => 'list' }
+    assert_response :success
+    puppetclasses = ActiveSupport::JSON.decode(@response.body)
+    assert !puppetclasses.empty?
```
Comment (dLobatog) [inline test/functional/api/v2/puppetclasses_controller_test.rb]: Nitpick, `assert_not_empty puppetclasses` or `assert_not puppetclasses.empty?` look easier to read for me.

**Comment by dLobatog** [comment]
:+1: except for the two snarky comments :smile: , will merge after that.

File: test/functional/api/v2/puppetclasses_controller_test.rb (hunk 2 above)
Comment (isratrade) [inline test/functional/api/v2/puppetclasses_controller_test.rb]: I chose `refute_empty` per the minitest style and removed the extra comment.

Comment (dLobatog) [inline test/functional/api/v2/puppetclasses_controller_test.rb]: I see `refute_empty` everywhere but on this test :wink:

**Comment by isratrade** [comment]
@eLobato, updated. no `!` in `test/functional/api/v2/puppetclasses_controller_test.rb` However, there are plentiful throughout the rest of the test suite.

**Comment by dLobatog** [comment]
Merged https://github.com/theforeman/foreman/commit/c95aebfbe524ebee89e32347c0051034eab3b611 , thanks @isratrade !

**Pull request closed without merging**